File: Data Structures & Algorithms/is-palindrome/submission-1.java
class Solution {
    public boolean isPalindrome(String s) {
        String str = s.replaceAll("[^a-zA-Z0-9]", "")
                        .toLowerCase();
        int left = 0;
        int right = str.length() - 1;

        while (left < right) {
            char a = str.charAt(left);
            char b = str.charAt(right);
            if (a != b) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        res = 0

        def dfs(node: Optional[TreeNode]) -> int:
            nonlocal res

            if not node:
                return 0
            
            leftDepth = dfs(node.left)
            rightDepth = dfs(node.right)

            res = max(res, leftDepth + rightDepth)

            return 1 + max(leftDepth, rightDepth)
        
        dfs(root)
        return res
        